### DISCUSSION TITLE
How to set http auth credentials for a site?

### DISCUSSION THREAD

**Discussion by phatpaul** [pyload/pyload#4269] [Help] [answered]
Do we really need a plugin for every download site?  Some private sites just use basic HTTP auth.  Seems to me like it should be possible to set my username/password for a private download site in the configuration of pyLoad.  (or in a generic plugin)

I.e. if I add a link from a configured domain, it should just add my credentials to the request as basic HTTP auth.
My settings:
[easynews.com]
domain=easynews.com
username=myusername
password=mypassword

Then adding a link like `https://easynews.com/downloadfile.txt` is translated to `https://myusername:mypassword@easynews.com/downloadfile.txt`

BTW, I did find [this plugin for my site,](https://github.com/angeloxx/pyload-easynews) but I couldn't figure out how to install it.  I wish pyLoad would just handle generic http auth without needing a plugin.

**Comment by GammaC0de** [comment, marked as answer]
You can either
* Use `https://myusername:mypassword@easynews.com/downloadfile.txt` as URL
* On the Add Package dialog you can add the URL without credentials and then add the credentials in the password text box in the form `myusername:mypassword`.
* Go to the accounts page and add a `Http` account: for username you put the site address (i.e. `easynews.com`) and for password you put the credentials in the form `myusername:mypassword`.

Cheers 🍺

**Reply by Ksrx01** [reply]
Sadly I was unable to get this to work. Apparently this way of authentication is also considered the least secure.

**Comment by phatpaul** [comment]
Thanks!  That was not obvious at all.  Maybe add a hint on the plugin's description of how to use it?

Does it work for wildcard subdomains?  For now I added several instances for various subdomains i.e. `https://members.easynews.com` and `https://dl04.easynews.com` etc...  But a wildcard would make this easier.

**Reply by GammaC0de** [reply]
Added wildcard  support.

**Comment by dansleboby** [comment]
Hi, it is a possibility to implement: https://requests.readthedocs.io/en/latest/user/authentication/#digest-authentication

Basicly if the website have a header WWW-Authenticate we need to send a digest auth.

This is uselful to download from some Webdav and other software.

Thanks
